io.quarkus.websockets.next.deployment.WebSocketProcessor fails to validate injection endpoint when @WebSocketClient class in library

### Describe the bug

I had tested a quarkus-websockets-next in a single project, but when I separtated out the `@WebSocketClient` class into a separate library, the service would no longer build.

### Expected behavior

Using a `@WebSocketClient` endpoint class from library should work

### Actual behavior

The service no longer builds

```bash
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.28.4:build (default) on project ws-service: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#generateResources threw an exception: jakarta.enterprise.inject.spi.DeploymentException: io.quarkus.websockets.next.WebSocketClientException: Type argument [com.si.prices.endpoint.MiniTickerStreamsEndpoint] of the injected WebSocketConnector is not a @WebSocketClient endpoint: com.si.prices.alerts.WsStreamConnectionService#streams
[ERROR] 	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1634)
[ERROR] 	at io.quarkus.arc.processor.BeanProcessor.processValidationErrors(BeanProcessor.java:198)
[ERROR] 	at io.quarkus.arc.deployment.ArcProcessor.generateResources(ArcProcessor.java:521)
[ERROR] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:735)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:874)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:255)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2651)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2630)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.runThreadBody(EnhancedQueueExecutor.java:1622)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1589)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1447)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] Caused by: io.quarkus.websockets.next.WebSocketClientException: Type argument [com.si.prices.endpoint.MiniTickerStreamsEndpoint] of the injected WebSocketConnector is not a @WebSocketClient endpoint: com.si.prices.alerts.WsStreamConnectionService#streams
[ERROR] 	at io.quarkus.websockets.next.deployment.WebSocketProcessor.validateConnectorInjectionPoints(WebSocketProcessor.java:449)
[ERROR] 	... 10 more
```


### How to Reproduce?

1. git clone https://github.com/starksm64/wsnext-bug
2. cd wsnext-bug
3. mvn verify


### Output of `uname -a` or `ver`

Darwin Scotts-Mac-Studio.local 24.6.0 Darwin Kernel Version 24.6.0: Mon Jul 14 11:30:29 PDT 2025; root:xnu-11417.140.69~1/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

openjdk 24.0.1 2025-04-15 OpenJDK Runtime Environment Temurin-24.0.1+9 (build 24.0.1+9) OpenJDK 64-Bit Server VM Temurin-24.0.1+9 (build 24.0.1+9, mixed mode, sharing)

### Quarkus version or git rev

3.28.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937) Maven home: /Users/starksm/bin/apache-maven-3.9.9 Java version: 24.0.1, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-24.jdk/Contents/Home Default locale: en_US, platform encoding: UTF-8 OS name: "mac os x", version: "15.6.1", arch: "aarch64", family: "mac

### Additional information

If you move the `@WebSocketClient` class into the same project as the class doing the `WebSocketConnector` injection, the project builds. See these lines in the `com.si.prices.alerts.WsStreamConnectionService` class. If the first injection site is commented out in favor of the second the project builds:

HackMTEndpoint

```java
    @Inject
    WebSocketConnector<MiniTickerStreamsEndpoint> streams;
    //@Inject 
    //WebSocketConnector<HackMTEndpoint> streams;
```

--- a/extensions/websockets-next/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketProcessor.java
+++ b/extensions/websockets-next/deployment/src/main/java/io/quarkus/websockets/next/deployment/WebSocketProcessor.java
@@ -437,19 +437,30 @@ public class WebSocketProcessor {
 
     @BuildStep
     void validateConnectorInjectionPoints(List<WebSocketEndpointBuildItem> endpoints,
+            BeanDiscoveryFinishedBuildItem beanDiscoveryFinishedBuildItem,
             ValidationPhaseBuildItem validationPhase, BuildProducer<ValidationErrorBuildItem> validationErrors) {
         for (InjectionPointInfo injectionPoint : validationPhase.getContext().getInjectionPoints()) {
             if (injectionPoint.getRequiredType().name().equals(WebSocketDotNames.WEB_SOCKET_CONNECTOR)
                     && injectionPoint.hasDefaultedQualifier()) {
                 Type clientEndpointType = injectionPoint.getRequiredType().asParameterizedType().arguments().get(0);
-                if (endpoints.stream()
-                        .filter(WebSocketEndpointBuildItem::isClient)
-                        .map(WebSocketEndpointBuildItem::beanClassName)
-                        .noneMatch(clientEndpointType.name()::equals)) {
+                if (beanDiscoveryFinishedBuildItem.beanStream().withBeanClass(clientEndpointType.name()).isEmpty()) {
+                    // there is no CDI bean with given name
                     validationErrors.produce(
                             new ValidationErrorBuildItem(new WebSocketClientException(String.format(
-                                    "Type argument [%s] of the injected WebSocketConnector is not a @WebSocketClient endpoint: %s",
+                                    "Type argument [%s] of the injected WebSocketConnector is not a @WebSocketClient endpoint, because it is not a bean: %s"
+                                            + "\nPlease consult https://quarkus.io/guides/cdi-reference#bean_discovery on how to make the module containing "
+                                            + "the code discoverable by Quarkus. ",
                                     clientEndpointType, injectionPoint.getTargetInfo()))));
+                } else {
+                    if (endpoints.stream()
+                            .filter(WebSocketEndpointBuildItem::isClient)
+                            .map(WebSocketEndpointBuildItem::beanClassName)
+                            .noneMatch(clientEndpointType.name()::equals)) {
+                        validationErrors.produce(
+                                new ValidationErrorBuildItem(new WebSocketClientException(String.format(
+                                        "Type argument [%s] of the injected WebSocketConnector is not a @WebSocketClient endpoint: %s",
+                                        clientEndpointType, injectionPoint.getTargetInfo()))));
+                    }
                 }
             }
         }

